src/NServiceBus.Persistence.DynamoDB/Installer.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace NServiceBus.Persistence.DynamoDB
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Amazon.DynamoDBv2;
+    using Amazon.DynamoDBv2.Model;
+    using Installation;
+    using Logging;
+
+    class Installer : INeedToInstallSomething
+    {
+        public Installer(IProvideDynamoDBClient clientProvider, InstallerSettings settings)
+        {
+            installerSettings = settings;
+            this.clientProvider = clientProvider;
+        }
+
+        public async Task Install(string identity, CancellationToken cancellationToken = default)
+        {
+            if (installerSettings == null || installerSettings.Disabled)
```

Comment:
Microsoft.Extensions.DependencyInjection would normally throw if the dependency is missing (unless the parameter has a default value) but the behavior can vary depending on the underlying container. For example, if DryIoc is used and exposed over the DI abstractions it can inject null :(